src/output.rs

```diff
@@ -121,6 +121,8 @@ pub trait OutputTypeDetection {
     fn is_opreturn_omni(&self) -> bool;
     fn is_opreturn_stacks_blockcommit(&self) -> bool;
     fn is_opreturn_with_len(&self, length: usize) -> bool;
+    fn is_opreturn_bip47_payment_code(&self) -> bool;
+    fn is_member_secp256k1(sign_byte: u8, x_xoordinate: &[u8]) -> bool;
```

Comment:
I don't think putting this function there makes sense. It's not an OP_RETURN flavor and probably doesn't need to be part of `OutputTypeDetection` at all?

Reply:
Agree. I don't find any other appropriate place (utils) to shift this fn to . Should i create a new module ? This being one off, i think its better to code the function at its call location
```suggestion
        // Bytes 3-34 - x value, must be a member of the secp256k1 group
        let x_coordinate = &payload[3..35];
        let mut pubkey_bytes = Vec::with_capacity(33);
        pubkey_bytes.push(sign_byte);
        pubkey_bytes.extend_from_slice(x_coordinate);
        if !secp256k1::PublicKey::from_slice(&pubkey_bytes).is_ok() {
           return false
        }
```